bundle/edu.gemini.pot/src/main/scala/edu/gemini/spModel/gemini/ghost/Ghost.scala

```diff
@@ -285,11 +314,43 @@ final class Ghost extends SPInstObsComp(GhostMixin.SP_TYPE) with PropertyProvide
   }
 
   override def getVignettableScienceArea: ScienceAreaGeometry = GhostScienceAreaGeometry
+
+  override def pwfs2VignettingClearance(ctx: ObsContext): Angle = {
+    val distance = if (ctx.getTargets == null)
+      Angle.ANGLE_0DEGREES
+    else {
+      // TODO-GHOST: It seems that there should be an easier way to do this but I had difficulty getting other
+      // TODO-GHOST: implementations to compile.
+      // List of all coordinates of science and sky positions.
+      val coordsList: List[Coordinates] = {
+        val asterism = ctx.getTargets.getAsterism
+        (asterism.allSpTargets.toList ++ asterism.allSpCoordinates).flatMap(_.getSkycalcCoordinates(ctx.getSchedulingBlockStart).asScalaOpt)
+      }
+
+      // Find all the coordinate differences from the base, if it is set.
+      val diffs = for {
+        base <- ctx.getBaseCoordinates.asScalaOpt.toList
+        coord <- coordsList
+      } yield new CoordinateDiff(base, coord).getDistance
+
+      // Get the maximum.
+      diffs.foldLeft(Angle.ANGLE_0DEGREES) { (a1, a2) => if (a1.compareToAngle(a2) > 0) a1 else a2 }
+    }
+
+    // Add the boundary distance as defined below in the companion object.
+    distance.add(Ghost.Pwfs2Boundary)
```

Comment:
VIGNETTING

Here we add the extra "boundary," which I currently don't know what it is. I was looking at the Pwfs2 guider code to get a sense of if I could determine it from there but measurements are in `mm`. Would experimentally determining this value be better, or should it be calculated? Again, is it a static value, or a dynamic one based on the `ObsContext`? I'm not sure here.

Reply:
This is all along the lines of what I had proposed but as we discussed I think my suggestion was not a good one in the end.  I don't see how we can do this calculation without taking into account the actual position of the probe arm, which means having the geometry in the model. Sorry to have steered you down a dead-end path like this!